homeassistant/components/mawaqit/coordinator.py

```diff
@@ -0,0 +1,114 @@
+"""Example integration using DataUpdateCoordinator."""
+
+from datetime import datetime, timedelta
+import logging
+
+from mawaqit.consts import BadCredentialsException, NoMosqueAround, NoMosqueFound
+
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+import homeassistant.util.dt as dt_util
+
+from . import mawaqit_wrapper
+from .const import CONF_UUID
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MosqueCoordinator(DataUpdateCoordinator):
+    """Coordinator to fetch mosque information."""
+
+    def __init__(self, hass: HomeAssistant, config_entry) -> None:
+        """Initialize the mosque coordinator."""
+        self.hass = hass
+        self.mosque_uuid = config_entry.data.get(CONF_UUID)
+        self.token = config_entry.data.get(CONF_API_KEY)
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Mosque Data",
+            update_method=self._async_update_data,
+            update_interval=timedelta(days=1),  # Updated every day
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch mosque details from local storage."""
+        try:
+            mosque_data = await mawaqit_wrapper.fetch_mosque_by_id(
+                self.mosque_uuid, token=self.token
+            )
+
+        except BadCredentialsException as err:
+            _LOGGER.error("Bad credentials: %s", err)
+            # Handle re-authentication if needed
+        except NoMosqueAround as err:
+            _LOGGER.error("No mosque found in the area: %s", err)
+        except NoMosqueFound as err:
+            _LOGGER.error("No mosque found: %s", err)
+        except (ConnectionError, TimeoutError) as err:
+            _LOGGER.error("Network-related error: %s", err)
+
+        if not mosque_data:
+            _LOGGER.error("No mosque data found from the API")
+            raise UpdateFailed("No mosque data found from the API")
+
+        return mosque_data
+
+
+class PrayerTimeCoordinator(DataUpdateCoordinator):
+    """Coordinator to fetch prayer times from the Mawaqit API."""
+
+    def __init__(self, hass: HomeAssistant, config_entry) -> None:
+        """Initialize the prayer time coordinator."""
+
+        self.hass = hass
+        self.mosque_uuid = config_entry.data.get(CONF_UUID)
+        self.token = config_entry.data.get(CONF_API_KEY)
+        self.last_fetch: datetime | None = None
+        self.prayer_times: dict | None = None
+
+        super().__init__(
+            hass,
+            _LOGGER,
+            name="Prayer Times",
+            update_method=self._async_update_data,
+            update_interval=timedelta(minutes=1),
+        )
+
+    async def _async_update_data(self) -> dict:
+        """Fetch prayer times from API, and notify sensors."""
```

Comment:
Please explain in the docstring that we hit the API once per day, but update the sensors every minute.